### ISSUE TITLE
Spurious 'lacks side effects' warning for comma expression with side effects

### ISSUE THREAD

**Issue by kg** [google/closure-compiler#475]
For some of the comma expressions in this file:
http://hildr.luminance.org/MonoGame/Platformer2D/MonoGame.Framework,%20Version=3.0.0.0,%20Culture=neutral,%20PublicKeyToken=null.js

Closure in simple mode produces warnings like this:

```
JSC_USELESS_CODE: Suspicious code. This code lacks side-effects. Is there a bug? at line 2050 character 8 in MonoGame.Framework,%20Version=3.0.0.0,%20Culture=neutral,%20PublicKeyToken=null.js
$temp00)) {
```

The relevant statement, however, clearly has side effects:

``` javascript
    var $temp00;

    for (var a$0 = this.disposableAssets._items, i$0 = 0, l$0 = this.disposableAssets._size; i$0 < l$0; ($temp00 = i$0, 
        i$0 = ((i$0 + 1) | 0), 
        $temp00)) {
```

Tested using http://closure-compiler.appspot.com/home.

To be specific, my codegen uses comma expressions to emulate the semantics of pre/post-increment while maintaining type information (asm.js style). So in this case, removing the comma expression would actually change the behavior of the code.

Alternately, if there's a way to change my codegen to suppress this warning, i'd be okay with that too.
